frontend/src/pages/projects/screens/detail/notebooks/NotebookTableRow.tsx

```diff
@@ -37,6 +37,12 @@ const NotebookTableRow: React.FC<NotebookTableRowProps> = ({
   const { size: notebookSize, error: sizeError } = useNotebookDeploymentSize(obj.notebook);
   const [notebookImage, loaded] = useNotebookImage(obj.notebook);
 
+  const imageDisplayName =
+    notebookImage?.imageName ||
+    obj.notebook.metadata.annotations['opendatahub.io/image-display-name'] ||
```

Comment:
Our types must be wrong because `annotations` is not a required field in k8s. It's possible to create a notebook via the cli or the console and not have any annotations.
```suggestion
    obj.notebook.metadata.annotations?.['opendatahub.io/image-display-name'] ||
```

Reply:
If types are wrong, this change needs to come with a corresponding type change too. K8s says no to a lot of things that contextually we may know more about the use-case. Maybe Notebooks always have annotations.

Since this is not something we can guarantee on our side, we probably should favour the state of stronger types as that's a runtime error waiting to happen.

Reply:
add the `?`

Reply:
@Gkrumbach07 
Yes add the `?` but also it comes back to the underlying type: `NotebookKind`. It also needs to be updated. `annotations` and `labels` are optional in k8s resources and should be reflected as such in the type.

Reply:
I made those optional too. should be updated